strutil/strutil.go

```diff
@@ -511,3 +512,37 @@ func Reverse(in string) string {
 	}
 	return string(out)
 }
+
+// ReplaceNonMatcher replaces all characters within a given that do not match
+// a given regular expression. The function returns the updated string and the
+// number of replacements
+func ReplaceNonMatcher(in, matcher string, replaceWith byte) (string, int) {
```

Comment:
It seems a bit odd to pass in a byte as the replacement character, since a string is represented by bytes but is actually a sequence of runes. So if you want a single character, you should probably pass in a rune.

Going a step further though, why not just allow passing in a string? A string of length one will act the same as this function, but you can replace with longer strings if you want.

Of course, if you do that, now you've just created https://pkg.go.dev/regexp#Regexp.ReplaceAllString and I do have to wonder if there is value in this function over just using that.

Reply:
Make sense to me to go for a rune/string as type for `replaceWith`. I will adjust that. 

re https://pkg.go.dev/regexp#Regexp.ReplaceAllString : This function is intended to be the reverse way. So like replacing everything that does not match the regex. Leveraging `Regexp.ReplaceAllString` will transform the code to be much smoother. Will work on that 👌